### DISCUSSION TITLE
Safari version support on iOS

### DISCUSSION THREAD

**Discussion by goranalkovic-infinum** [infinum/eightshift-frontend-libs#437] [General]
As we usually support _last 2 versions_ of browsers, a little clearer definition would be nice regarding iOS.

As Safari releases are tied with iOS releases, it's not uncommon for minor versions of iOS to include different minor (or sometimes even major) versions of Safari.

This is important as sometimes feature support is fragmented and in some cases very broken depending on Safari versions.

Here's a table with the last 7 minor versions, their usage stats and Safari version (data is from Statscounter from August 2021):

| iOS version   | Released                                        | Safari version | Usage (worldwide) | Usage (Europe) | Usage (USA) |
|---------------|-------------------------------------------------|----------------|-------------------|----------------|-------------|
| 12.x, 13.x \*        | 13.x: 09/2019 - 09/2020, 12.x: 09/2018 - 09/2019 | 12.x, 13.x           | 14.7%             | 13.31%         | 9.1%        |
| 14.0          | 09/2020                                         | 14.0           | 0.9%              | 0.78%          | 0.51%       |
| 14.1          | 10/2020                                         | 14.0           | 0.6%              | 0.41%          | 0.47%       |
| 14.2          | 11/2020                                         | 14.0           | 1.87%             | 1.18%          | 2.04%       |
| 14.3          | 12/2020                                         | 14.0           | 1.47%             | 0.98%          | 1.3%        |
| 14.4          | 01/2021                                         | 14.0           | 7.72%             | 7.15%          | 6.55%       |
| 14.5          | 04/2021                                         | 14.1           | 1.53%             | 1.2%           | 1.2%        |
| 14.6          | 05/2021                                         | 14.1           | 30.18%            | 32.58%         | 32.16%      |
| 14.7          | 07/2021                                         | 14.1           | 40.63%            | 41.98%         | 46.25%      |
| 14.8          | 09/2021                                         | 14.1           | ? \*\*           | ? \*\*        | ? \*\*      |
| 15.0          | in Beta (release most likely in 09/2021)        | 15             | 0.4%              | 0.43%          | 0.42%       |
|||||||
| >= 14.5 total | >= 04/2021                                      | >= 14.1        | 72.74%            | 76.19%         | 80.03%      |

\* mostly 13.x
\*\* this version is too new to have stats at the time of writing

Please not that this table will already be out of date very soon, as today (Sep 14 2021) iOS 15 is expected to be released so its usage will increase relatively quickly.

From Safari 14.0 to 14.1 there weren't many new additions, but there were some bugfixes, and simple things that can be very widely used like flexbox gap are supported from there.

Safari 15.0 adds `aspect-ratio`, but we'll have to be more patient for that just yet 😄 

For desktop we should be fine, as Safari 14.1 is already supported on the last couple major macOS versions and Safari 15 is coming in the next macOS release.

Of course, nothing is perfect, there are many bugs with basic features in the newest couple of Safari versions, but we could avoid a bit of redundancy and duplication for some things, which would lighten and speed up some pages.

How feasible is it for us to support the last major version and X minor versions of the major version before it?
For example
- iOS 15.x
- iOS 14.5+

**Comment by mbmjertan** [comment]
[Apple's App Store metrics](https://developer.apple.com/support/app-store/) say that 90% of devices released in the last four years are on iOS 14, while 85% of all active iPhones use iOS 14. 

Note that iOS has around [26.4% market share worldwide](https://gs.statcounter.com/os-market-share/mobile/worldwide) and [56.7% market share in the US](https://gs.statcounter.com/os-market-share/mobile/united-states-of-america). 

Dropping support for iOS < 14.5 would mean dropping support for 7.1% of phones worldwide or up to 11.3% in the US.

Having all of these figures in mind, it seems okay (in the "not great, not terrible" sense) to support only the last major version of iOS (and perhaps a few minor releases before it), and I'd suggest considering `iOS x.5` releases as major versions: they have great adoption and usually bring enough features to consider them major. This would mean we'd support iOS 14.5+ at this point, or perhaps iOS 14.3+.
